sources/lib/util/options.ml

```diff
@@ -381,7 +391,34 @@ let mk_execution_opt frontend input_format parse_only parsers
     preludes no_locs_in_answers type_only type_smt2
   =
   let answers_with_loc = not no_locs_in_answers in
-  let input_format = "." ^ input_format in
+
+  vinfer_input_format :=
+    (match input_format with Some _ -> false | None -> true);
+  let input_format = match input_format with
+    | None -> INative
+    | Some fmt when match_format fmt = "native" -> INative
+    | Some fmt when match_format fmt = "smtlib" -> ISmtlib
+    | Some fmt when match_format fmt = "why3" ->
+      let why3 = Str.regexp_case_fold "why3" in
+      let found =
+        List.exists (fun parser ->
+            try
+              let _ = Str.search_forward why3 parser 0 in
+              true
+            with Not_found -> false
+          ) parsers
+      in
+      if not found then
+        let m = "Why3 format is only available when AB-Why3 parser is loaded"
+        in
+        raise (Error (false, m));
+      else
+        IWhy3
+    (*     | Some fmt when match_format fmt = "szs" -> ISZS *)
+    | Some fmt ->
+      let m = "Option --input does not accept the argument \"" ^ fmt in
+      raise (Error (false, m))
```

Comment:
Please, please use cmdliner's already existing mechanism for parsing arguments.
Basically, this would mean splitting this function in two: one for parsing strings into the `input` type, which ideally would be used only to create a cmdliner's argument converter (see https://erratique.ch/software/cmdliner/doc/Cmdliner.Arg.html#VALconv ), and another function, the setter for the argument which could check the availability of parsers.
Furthermore, use of the `Str` module is at best not recommanded or even deprecated, particularly in this case, where additionally trying to find a parser with "why3" in its name is a particularly atrocious and a brittle hack.
Overall, this brings about the following point:
- either we use a regular ADT to represent input formats, in which case it pretty much precludes the possibility of registering custom parsers for languages not in the definition of the ADT. In this case, why3 would be pretty much the only reasonable use of a custom parser (and we could maybe load it automatically ?)
- or, we use an extensible variant type for input_formats, but this brings about quite a lot of pain to the mechanism which would parse the input format argument string into this extensible variant
- or, if we really must allow and support custom parsers for custom formats (which I somehow doubt given the complexity of producing a parsed AST that would then be correctly typed), well... we can discuss other options.